include/sviper/merge_split_alignments.h

```diff
@@ -125,19 +134,11 @@ inline void transform_cigar_right_into_insertion(seqan::BamAlignmentRecord & rec
                      record.cigar,
                      [&until] (int ref, int /*read*/) {return ref >= until;});
 
-//    if (cigar_pos == 0) // no advancement happened. only case: until=1
-//    {
-//        seqan::erase(record.cigar, cigar_pos + 1, length(record.cigar)); // seqan::erase the rest
-//        record.cigar[0].count = until;
-//        return;
-//    }
+    SEQAN_ASSERT_NEQ(cigar_pos, 0u); // no advancement would have happened. edge case: until=1?
     --cigar_pos; // move to cigar operation that over stepped
     unsigned insertion_size{0};
 
-    if (ref_pos == until)
-    {
-        --cigar_pos;
-    }
+    // if (ref_pos == until) we don't need to do anything
     if (ref_pos > until)
```

Comment:
If `ref_pos > until` we have an overlap otherwise not?

Reply:
yes! :)